chromium_src/components/bookmarks/browser/bookmark_model_unittest.cc

```diff
@@ -0,0 +1,143 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#define NodeVisibility DISABLED_NodeVisibility
+#define ParentForNewNodes DISABLED_ParentForNewNodes
+#define MostRecentlyModifiedFolders DISABLED_MostRecentlyModifiedFolders
+#include "../../../../../components/bookmarks/browser/bookmark_model_unittest.cc"
+#undef MostRecentlyModifiedFolders
+#undef ParentForNewNodes
+#undef NodeVisibility
+namespace bookmarks {
+
+TEST_F(BookmarkModelTest, BraveNodeVisibility) {
+  // Mobile node invisible by default
+  EXPECT_TRUE(model_->bookmark_bar_node()->IsVisible());
+  // Other node invisible by default
+  EXPECT_FALSE(model_->other_node()->IsVisible());
+  EXPECT_FALSE(model_->mobile_node()->IsVisible());
+
+  // Visibility of permanent node can only be changed if they are not
+  // forced to be visible by the client.
+  model_->SetPermanentNodeVisible(BookmarkNode::BOOKMARK_BAR, false);
+  EXPECT_TRUE(model_->bookmark_bar_node()->IsVisible());
+  // Other node is persistently invisible
+  model_->SetPermanentNodeVisible(BookmarkNode::OTHER_NODE, true);
+  EXPECT_FALSE(model_->other_node()->IsVisible());
+  model_->SetPermanentNodeVisible(BookmarkNode::MOBILE, true);
+  EXPECT_TRUE(model_->mobile_node()->IsVisible());
+  model_->SetPermanentNodeVisible(BookmarkNode::MOBILE, false);
+  EXPECT_FALSE(model_->mobile_node()->IsVisible());
+
+  // Arbitrary node should be visible
+  TestNode bbn;
+  PopulateNodeFromString("B", &bbn);
+  const BookmarkNode* parent = model_->mobile_node();
+  PopulateBookmarkNode(&bbn, model_.get(), parent);
+  EXPECT_TRUE(parent->children().front()->IsVisible());
+
+  // Mobile folder should be visible now that it has a child.
+  EXPECT_TRUE(model_->mobile_node()->IsVisible());
+
+  // Even other node has a child, it remains invisible
+  TestNode node_a;
+  PopulateNodeFromString("A", &node_a);
+  PopulateBookmarkNode(&node_a, model_.get(), model_->other_node());
+  EXPECT_TRUE(model_->other_node()->children().front()->IsVisible());
+  EXPECT_FALSE(model_->other_node()->IsVisible());
+}
+
+// Tests that adding a URL to a folder updates the last modified time.
```

Comment:
Why the comment says about `the last modified time`?

Reply:
ditto